dace/codegen/targets/rtl.py

```diff
@@ -0,0 +1,544 @@
+            for var_name in tasklet.in_connectors
+        ])
+        return "// initialize vector\n" if len(
+            init_vector_string) > 0 else "" + init_vector_string
+
+    def generate_cpp_num_elements(self):
+        # TODO: compute num_elements=#elements that enter/leave the pipeline, for now we assume in_elem=out_elem (i.e. no reduction)
+        return "int num_elements = {};".format(1)
+
+    def generate_cpp_internal_state(self, tasklet):
+        internal_state_str = " ".join([
+            "{}=0x%x".format(var_name) for var_name in {
+                **tasklet.in_connectors,
+                **tasklet.out_connectors
+            }
+        ])
+        internal_state_var = ", ".join([
+            "model->{}".format(var_name) for var_name in {
+                **tasklet.in_connectors,
+                **tasklet.out_connectors
+            }
+        ])
+        return internal_state_str, internal_state_var
+
+    def unparse_tasklet(self, sdfg: SDFG, dfg: StateSubgraphView, state_id: int,
+                        node: nodes.Node, function_stream: CodeIOStream,
+                        callsite_stream: CodeIOStream):
+
+        # extract data
+        state = sdfg.nodes()[state_id]
+        tasklet = node
+
+        # construct variables paths
+        unique_name: str = "top_{}_{}_{}".format(sdfg.sdfg_id,
+                                                 sdfg.node_id(state),
+                                                 state.node_id(tasklet))
+
+        # generate system verilog module components
+        parameter_string: str = self.generate_rtl_parameters(sdfg.constants)
+        inputs, outputs = self.generate_rtl_inputs_outputs(sdfg, tasklet)
+
+        # create rtl code object (that is later written to file)
+        self.code_objects.append(
+            CodeObject(
+                name="{}".format(unique_name),
+                code=RTLCodeGen.RTL_HEADER.format(name=unique_name,
+                                                  parameters=parameter_string,
+                                                  inputs="\n".join(inputs),
+                                                  outputs="\n".join(outputs)) +
+                tasklet.code.code + RTLCodeGen.RTL_FOOTER,
+                language="sv",
+                target=RTLCodeGen,
+                title="rtl",
+                target_type="",
+                additional_compiler_kwargs="",
+                linkable=True,
+                environments=None))
+
+        # generate verilator simulation cpp code components
+        inputs, outputs = self.generate_cpp_inputs_outputs(tasklet)
+        vector_init = self.generate_cpp_vector_init(tasklet)
+        num_elements = self.generate_cpp_num_elements()
+        internal_state_str, internal_state_var = self.generate_cpp_internal_state(
+            tasklet)
+
+        # add header code to stream
+        sdfg.append_global_code(cpp_code=RTLCodeGen.CPP_HEADER_TEMPLATE.format(
+            name=unique_name))
```

Comment:
Please:
* Split to two headers (general and model headers)
* Check if you already printed out the general header and don't print it twice (you can add a field in `self` for that)